plugin/src/rules/sort_imports.rs

```diff
@@ -0,0 +1,687 @@
+use std::{borrow::Cow, sync::Arc};
+
+use grouped_ordering::grouped_ordering;
+use itertools::Itertools;
+use serde::Deserialize;
+use squalid::{EverythingExt, OptionExt};
+use tree_sitter_lint::{
+    range_between_start_and_end, rule, tree_sitter::Node, tree_sitter_grep::SupportedLanguage,
+    violation, NodeExt, QueryMatchContext, Rule, SourceTextProvider,
+};
+
+use crate::{
+    assert_kind,
+    ast_helpers::get_num_import_specifiers,
+    kind::{
+        Identifier, ImportClause, ImportSpecifier, ImportStatement, NamedImports, NamespaceImport,
+    },
+};
+
+grouped_ordering!(MemberSyntaxSortOrder, [None, All, Multiple, Single,]);
```

Comment:
I thought this pattern of "an array of known "groups" that must include all of the groups" was interesting from both a deserialization and type-theoretic perspective so took a stab at abstracting it (including a `grouped_ordering!()` macro to generate most of the code)